### PR TITLE
Add support on machines where long double and float128 do not have the same size

### LINKED ISSUE #118: Support for `ARM64` architectures that have `long double` as 64bit
Currently, we do not have proper support for any architecture were `float128` does not have the same size & alignment as `long double`. Unfortunately, this includes Apple's M1 architecture, which is gaining popularity as Mac users update their machines.

We should thus:
- [x] Expose `float128` as a distinct `fkind` on machines that have it.

This is urgently needed as we have some students with recent versions of OS X in the practical course at TUM.

### PR DESCRIPTION
**Pull request by michael-schwarz**
Closes #118

### CODE REVIEW

**Comment by michael-schwarz** [comment]
Confirmed on newer machines, but somehow still produces warnings about unmatched cases somewhere, I'll have to investigate that.

File: src/cil.ml [hunk 1]
```diff
@@ -1844,9 +1852,11 @@ let d_const () c =
          FFloat -> chr 'f'
        | FDouble -> nil
        | FLongDouble -> chr 'L'
+       | FFloat128 -> chr 'L' (* TODO: correct? *)
```
Comment (sim642) [inline src/cil.ml]: https://gcc.gnu.org/onlinedocs/gcc/Floating-Types.html says:

> Use a suffix ‘q’ or ‘Q’ for `__float128`.

File: src/frontc/cabs2cil.ml [hunk 2]
```diff
@@ -3681,7 +3677,9 @@ and doExp (asconst: bool)   (* This expression is used as a constant *)
             (* Maybe it ends in U or UL. Strip those *)
             let l = String.length str in
             let baseint, kind =
-              if hasSuffix str "L" then
+              if hasSuffix str "F128" then
```
Comment (sim642) [inline src/frontc/cabs2cil.ml:3680]: Where is this `F128` taken from? GCC docs linked above say `q` or `Q` from what I can see.

File: src/frontc/cabs2cil.ml [hunk 3]
```diff
@@ -3711,7 +3709,9 @@ and doExp (asconst: bool)   (* This expression is used as a constant *)
             (* Maybe it ends in U or UL. Strip those *)
             let l = String.length str in
             let baseint, kind =
-              if hasSuffix str "iL" || hasSuffix str "Li" then
+              if hasSuffix str "iF128" || hasSuffix str "F128i" then
```
Comment (sim642) [inline src/frontc/cabs2cil.ml:3714]: Same `F128` question here.

File: src/frontc/cabs2cil.ml (hunk 2 above)
Comment (michael-schwarz) [reply src/frontc/cabs2cil.ml:3680]: At the very top of the link you found, it gives the suffixes for the `_FloatN` type. I'll have to check if the preprocessor turns `q` into this as well.

File: src/cil.ml (hunk 1 above)
Comment (michael-schwarz) [reply src/cil.ml]: Nice catch, `L` is wrong here for sure!

**Review by sim642 — APPROVED** [review]

**Pull request merged**